### DISCUSSION TITLE
Input form in TerriaJS

### DISCUSSION THREAD

**Discussion by Juanezm** [TerriaJS/terriajs#5121] [Q&A]
Hi all,

I'm currently trying to figure out how to integrate a REST API that produces data products using Open Data Cube (ODC) and later on can be visualised in TerriaJS.

The use case would be an input form available in the catalog with some input and drop down boxes that is sent to the REST API and this one returns some information about the requested product. When this request has been processed by ODC the new generated product would be stored in a S3 bucket and ready to be added to the served through geoserver.

The closest I could find is the TerriaJS JSON catalog function https://docs.terria.io/guide/connecting-to-data/catalog-type-details/terria-json/ but it only uses HTTP GET and I would need POST too.

**Comment by meh9** [comment]
WPS allows you to do those things and use POST, e.g. have a look at the GEOGLAM WPS functions:

https://map.geo-rapp.org/#share=s-4DDvSNgUipkaizQBeRUmhAzKLa1

https://docs.terria.io/guide/connecting-to-data/catalog-type-details/wps/
